app/main.py

```diff
@@ -44,3 +69,45 @@ def main():
 def show_records(db: Session = Depends(get_db)):
     records = db.query(models.EnergyType).all()
     return records
+
+@app.get("/energy-profile/")
+def get_profile_from_postal(postal_code: Optional[str] = None, province: Optional[str] = None, db: Session = Depends(get_db)):
+    records = {}
+    # get postal code and province from energy_profiles depending on input parameters
+    try:
+        if province != None:
+            records['province_name'] = db.query(models.Province).\
+                filter(models.Province.abbreviation == province.upper()).first().name
+            records['energy_profile_code'] = db.query(models.Province).\
+                filter(models.Province.abbreviation == province.upper()).first().energy_profile_code
+        elif postal_code != None and province == None:
+            records['energy_profile_code'] = db.query(models.Province).\
+                filter(models.Province.province_id == postal_to_province[postal_code[0].upper()]).\
+                    first().energy_profile_code
+            records['province_name'] = db.query(models.Province).\
+                filter(models.Province.province_id == postal_to_province[postal_code[0].upper()]).\
+                    first().name
+        else:
+            return {"ERROR":"no data"}
```

Comment:
We would probably want to throw a real error message / error code but this is fine for now since it doesn't really matter